kani-compiler/src/codegen_cprover_gotoc/codegen/assert.rs

```diff
@@ -50,11 +103,15 @@ impl<'tcx> GotocCtx<'tcx> {
         message: &str,
         loc: Location,
     ) -> Stmt {
-        assert!(cond.typ().is_bool());
         let property_name = property_class.as_str();
         Stmt::assert(cond, property_name, message, loc)
     }
 
+    /// Generates a CBMC assumption.
+    pub fn codegen_assume(&self, cond: Expr, loc: Location) -> Stmt {
```

Comment:
Yes: Several of these don't actually use `self` but some could in the future. I thought about this and decided that it might be best to keep them all regular. (In fact, the weirdest one was introducing `codegen_assume`, but I felt that having NO references to the `cprover_bindings` assert/assume/etc stuff outside of this file was a consistency worth having...)

Reply:
You could also just use pure functions without the `self`. I do agree that for consistency sake, we might as well not use `Stmt` directly.